src/marten/core/sluggable.cr

```diff
@@ -0,0 +1,31 @@
+module Marten
+  module Core
+    # The Sluggable module provides functionality for generating URL-friendly slugs from strings.
+    module Sluggable
+      extend self
+
+      NON_ALPHANUMERIC_RE  = /[^\w\s-]/
+      WHITESPACE_HYPHEN_RE = /[-\s]+/
+      NON_ASCII_RE         = /[^\x00-\x7F]/
```

Comment:
What about unicode characters? Ultimately, and specifically for certain locales, generating slug values containing unicode characters makes a lot of sense. Is it something that we should support?

Reply:
We could do that. But does the router support that? I just made a quick test with a route "/ü" and the router could not find it because it searched for the encoded "/%C3%BC".

I would say this could be done in a seperated issue if someone really needs it?